server/src/main/java/org/candlepin/model/AbstractHibernateCurator.java

```diff
@@ -171,6 +170,49 @@ public E get(Serializable id) {
         return id == null ? null : this.get(entityType, id);
     }
 
+    /**
+     * Checks if entry exists in db.
+     * @param id primary key of entity.
+     * @return boolean value whether row exists or not
+     */
+    @Transactional
+    public boolean exists(Serializable id) {
+        String columnName = null;
+        boolean doesExists = false;
+        columnName = getPrimaryColumnName();
+
+        if (columnName != null) {
+            String jpql = "SELECT count(t) FROM " + entityType.getCanonicalName() +
+                " t WHERE " + columnName + " = :id";
+
+            if (this.getEntityManager().createQuery(jpql, Long.class)
+                .setParameter("id", id)
+                .getSingleResult() > 0) {
+                doesExists = true;
+            }
+        }
+
+        return doesExists;
+    }
+
+    /**
+     * Returns name of primary key column
+     * @return Returns entity primary key column
+     */
+    public String getPrimaryColumnName() {
```

Comment:
We should follow Hibernate/JPA or SQL naming conventions here: either ```getIdentifierName``` or ```getPrimaryKeyName```.

Reply:
done.